Currency.ts

```diff
@@ -553,6 +375,33 @@ export namespace Currency {
 		const factor = Math.pow(10, decimalDigits(currency) ?? 2)
 		return Math.round((value + Number.EPSILON) * factor) / factor
 	}
+
+	export function add(currency: Currency, value1: number, value2: number): number {
+		const decimals = decimalDigits(currency)
+		const factor = Math.pow(10, decimals == 0 ? 2 : decimals ?? 2)
+		const result = Math.round(factor * value1 + factor * value2) / factor
+		return decimals == 0 ? Math.round(result) : result
+	}
+	export function divide(currency: Currency, nominator: number, denominator: number): number {
+		const decimals = decimalDigits(currency)
+		const factor = Math.pow(10, decimals == 0 ? 2 : decimals ?? 2)
+		const result = Math.round(factor * (nominator / denominator)) / factor
+		return decimals == 0 ? Math.round(result) : result
+	}
+	export function multiply(currency: Currency, value1: number, value2: number): number {
```

Comment:
Don't muliply currency with currency. Only convert value1 to minor units. Same for division.